### PR TITLE
Practice branch

### PR DESCRIPTION
**Pull request by damianololo**
my proposal current_weather_page UI

### CODE REVIEW

File: lib/weather/presentation/current_weather_page.dart [hunk 1]
```diff
@@ -17,37 +22,182 @@ class _WeatherGeneralState extends ConsumerState<CurrentWeatherPage> {
...
-              child: const Text("Throw Test Exception"),
-            ),
+        extendBodyBehindAppBar: true,
+        appBar: AppBar(
+          backgroundColor: Colors.transparent,
+          elevation: 0,
+          leading: Image.asset('lib/assets/images/App Logo.png'),
+          actions: [
+            IconButton(
+                onPressed: () => {}, icon: const Icon(Icons.density_medium))
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page.dart]: trailing coma `),)` then reformat

File: lib/weather/presentation/current_weather_page.dart [hunk 2]
```diff
@@ -17,37 +22,182 @@ class _WeatherGeneralState extends ConsumerState<CurrentWeatherPage> {
...
+                            image: DecorationImage(
+                                image: AssetImage(
+                                    'lib/assets/images/Rectangle 217.png'),
+                                fit: BoxFit.cover),
+                          ),
+                          height: 600,
+                          width: 300,
+                        ),
+                        Container(
+                          alignment: const Alignment(1, 0.1),
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page.dart]: I didn't observe any impact of the Alignment on UI

File: lib/weather/presentation/current_weather_page_components/small_table_stats.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+import 'package:yet_another_weather_app/weather/domain/current_weather_model.dart';
+
+class SmallTableStats extends StatelessWidget {
+  CurrentWeatherModel item;
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page_components/small_table_stats.dart]: You don't need to pass whole the object. You only use `humidity` so this is enough. It's a common mistake, which often creates spaghetti code in big projects (I experienced it hard, don't recommend)

Make it `final` please! It's inside a statelesswidget. You also can tell me in PM why final :)

**Review by Swirastlynn — CHANGES_REQUESTED** [review]
you should extract more widgets, and refactor of a Stack is probably a good idea

File: lib/weather/presentation/current_weather_page.dart (hunk 2 above)
Comment (damianololo) [reply lib/weather/presentation/current_weather_page.dart]: removing unnecessary code

File: lib/weather/presentation/current_weather_page_components/small_table_stats.dart (hunk 3 above)
Comment (damianololo) [reply lib/weather/presentation/current_weather_page_components/small_table_stats.dart]: I use humidity and windSpeed. Also I should use precipitation but i didn't see the precipitation property in the model

File: lib/weather/presentation/current_weather_page.dart [hunk 4]
```diff
@@ -22,182 +22,114 @@ class _WeatherGeneralState extends ConsumerState<CurrentWeatherPage> {
   Widget build(BuildContext context) {
     AsyncValue<CurrentWeatherModel> state =
         ref.watch(currentWeatherControllerProvider);
-    DateTimeHelper dateTimeHelper = DateTimeHelper();
-    String formattedDateTime = dateTimeHelper.getFormattedDateTime();
 
     // FIXME: UI layer. The task is to change this to a more appropriate widget.
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page.dart]: remove this line

File: lib/weather/presentation/current_weather_page_components/main_stats_view.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import 'package:yet_another_weather_app/weather/presentation/current_weather_page_components/date_time_helper.dart';
+import 'package:yet_another_weather_app/weather/presentation/current_weather_page_components/wind_calculator.dart';
+
+class MainStatsView extends StatelessWidget {
+  final double? temperature;
+  final double? windSpeed;
+  final int? cloudiness;
+  final DateTimeHelper dateTimeHelper = DateTimeHelper();
+  final WindCalculator windCalculator = WindCalculator();
+  final CloudCalculator cloudCalculator = CloudCalculator();
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page_components/main_stats_view.dart]: 1. 
```
  final DateTimeHelper dateTimeHelper = DateTimeHelper();
  final WindCalculator windCalculator = WindCalculator();
  final CloudCalculator cloudCalculator = CloudCalculator();
```
These 3 lines of code should be injected. 

By applying dependency injection, you could pass instances of the abstractions into the MainStatsView class through its constructor, allowing for better decoupling and testability. This way, the MainStatsView class can work with any implementation of the abstractions, promoting more modular and maintainable code.

2. Actually classes like DateTimeHelper, WindCalculator and CloudCalculator are not part of widgets. They are typical business logic, so they should be hidden from widgets. I'd rather use values calculated by them in the widget constructor. And not even care about dependency injection here. Rather somewhere they come from.

Comment (Swirastlynn) [reply lib/weather/presentation/current_weather_page_components/main_stats_view.dart]: 3. DateTimeHelper etc are actually utility classes. In Dart there's no need for Utility classes like in Java (or probably C#). Use global function usually, extension method from time to time.

Comment (Swirastlynn) [reply lib/weather/presentation/current_weather_page_components/main_stats_view.dart]: 4.
Actually, the clean solution is to use global functions for transforming cloudiness, windRating and other abstract values to String. How?

Introduce new Model named (for instance) CurrentWeatherPresentationModel. 
```
  final String formattedDateTime;
  final String iconCode;
  final String temperature;
  final String pressure;
  final String humidity;
  final String windSpeed;
  final String cloudiness;
```
This should be sent from Controller to Widget. 
What should be sent from UseCase to the Controller? CurrentWeatherModel. Transformations should be done in Controller.

File: lib/weather/presentation/current_weather_page_components/small_table_stats.dart [hunk 6]
```diff
@@ -1,11 +1,14 @@
 import 'package:flutter/material.dart';
-import 'package:yet_another_weather_app/weather/domain/current_weather_model.dart';
 
 class SmallTableStats extends StatelessWidget {
-  CurrentWeatherModel item;
-  SmallTableStats({
+  final int? pressure;
+  final int? humidity;
+  final double? windSpeed;
```
Comment (Swirastlynn) [inline lib/weather/presentation/current_weather_page_components/small_table_stats.dart]: There should be Strings there, not abstract values. 
This line of code `pressure?.toString() ?? "empty",` is part of business logic.

**Review by Swirastlynn — CHANGES_REQUESTED** [review]
My review may look like there's a lot of things to do yet, but this is the most important Code Review I have done since some time. Fixing it will teach you a lot.

**Pull request closed without merging**